docs/src/content/docs/utilities/add-virtual-import.mdx

```diff
@@ -0,0 +1,57 @@
+---
+title: addVirtualImport
+description: Allows adding Vite Virtual Modules easily to access data across your integration.
+---
+
+`addVirtualImport` allows you to create modules and store state across your integration. For example:
+
+```ts title="integration/index.ts"
+import type { AstroIntegration } from "astro";
+import { addVirtualImport } from "astro-integration-kit";
+
+export default function myIntegration(): AstroIntegration {
+    return {
+        name: "my-integration",
+        hooks: {
+            "astro:config:setup": ({ updateConfig }) => {
+                addVirtualImport({
+                    name: 'my-integration/config', // this will now be importable with `virtual:my-integration/config`
+                    content: {
+                        foo: 'bar',
+                    },
+                    updateConfig,
+                })
+            }
+        }
+    }
+}
+```
+
+So to read this config anywhere else in your integration is easy:
```

Comment:
May be worth specifying that virtual imports can't be used from within the integration itself